lib/pbench/agent/results.py

```diff
@@ -9,97 +9,70 @@
-                self.result_dir = Path(result_dir).resolve(strict=True)
-            except Exception:
+                if dir_name == "Result":
+                    self.result_dir = Path(dir_path).resolve(strict=True)
+                else:
+                    self.target_dir = Path(dir_path).resolve(strict=True)
+            except Exception as exc:
                 raise FileNotFoundError(
-                    "Invalid result directory provided: {}", result_dir
+                    f"Invalid {dir_name} directory provided: {dir_path}, ({exc})"
                 )
             else:
-                if not self.result_dir.is_dir():
-                    self.logger.error(
-                        "Invalid result directory provided: {}", result_dir
-                    )
-                    return 1
-
-        if not target_dir:
-            raise FileNotFoundError("Target directory not provided")
-        else:
-            try:
-                self.target_dir = Path(target_dir).resolve(strict=True)
-            except Exception:
-                raise FileNotFoundError(
-                    "Invalid target directory provided: {}", target_dir
-                )
-            else:
-                if not self.target_dir.is_dir():
-                    self.logger.error(
-                        "Invalid target directory provided: {}", target_dir
+                if not Path(dir_path).is_dir():
+                    raise NotADirectoryError(
+                        f"Invalid {dir_name} directory provided: {dir_path}"
                     )
-                    return 1
-        return 0
 
     def make_result_tb(self):
         if os.path.exists(f"{self.result_dir}.copied"):
-            self.logger.debug("Already copied {}", self.result_dir)
-            return (0, "")
+            raise Exception(f"Already copied {str(self.result_dir)}")
 
         pbench_run_name = self.result_dir.name
         if os.path.exists(f"{self.result_dir}/.running"):
-            self.logger.warning(
-                "The benchmark is still running in {} - skipping; if that is"
-                " not true, rmdir {}/.running, and try again",
-                pbench_run_name,
-                pbench_run_name,
+            raise RuntimeError(
+                f"The benchmark is still running in {pbench_run_name} - skipping; if that is"
+                f" not true, rmdir {pbench_run_name}/.running, and try again",
             )
-            return (0, "")
 
         mdlog_name = self.result_dir / "metadata.log"
         mdlog = ConfigParser()
         try:
             with mdlog_name.open("r") as fp:
                 mdlog.read_file(fp)
-        except FileNotFoundError:
-            self.logger.debug(
-                "The {}/metadata.log file seems to be missing", pbench_run_name
+        except Exception as exc:
+            raise FileNotFoundError(
+                f"The {pbench_run_name}/metadata.log file seems to be missing, ({exc})"
             )
```

Comment:
Y'know what?  This is great advice (if I do say so myself...and @dbutenhof agrees, of course...), but it turns out that it is totally unnecessary -- [Python does this for you](https://docs.python.org/3/tutorial/errors.html#exception-chaining):

> Exception chaining happens automatically when an exception is raised inside an except or finally section

(unless you specify `from None`).

So, I think this means that we only need to use `from` when we have code which catches, saves, and then _later_ raises an exception (or in the rare occurrences where we want to elide, break, or otherwise mess with the chain).

Reply:
Ok.. Thanks :)

Reply:
Done